hrm-service/src/contact-job/contact-job-complete.ts

```diff
@@ -73,6 +75,76 @@ const processCompletedContactJob = async (completedJob: CompletedContactJobBody)
   }
 };
 
+export const getAttemptNumber = async (completedJob: CompletedContactJobBody, contactJob: ContactJobRecord) => {
+  // Attempt number can be null when error occurs in lambda before job data is parsed
+  // In this case we need to fetch the job from the database to get the attempt number
+  if (completedJob.attemptNumber != null) {
+    return completedJob.attemptNumber;
+  }
```

Comment:
If `!=` is intended as a weak comparison for nullish values, I suggest using something like `Boolean(attemptNumber)`. If we want to only compare against `null`, let's change to `!==`?
https://stackoverflow.com/a/8616501

Reply:
It is meant to capture nullish values. JavaScript nullish comparisons are so confusing. As I understand it, that will capture only `null` or `undefined` where `Boolean(completedJob.attemptNumber)` or `!!completedJob.attemptNumber` would also capture `0`. But maybe best to use `??` based assignment which is a little more readable would make sense?

See [TSPlayground example](https://www.typescriptlang.org/play?#code/MYewdgzgLgBFCm0YF4YAoBuBDANgV3gC4YswBPASmICEQQd5SUA+GbfeGAQlTDxxwBYAFAjQkevAB0OEAHM0AAz4DiAEgDeCaGhU4KAX0UUx4CJJnyleMABN4AMwCWYeLfVbEUNDfvPXtobGIqYSDJYKivAAtgAOUGQw0ABOLnIe2t4A5FlBJsLi5uGykQAMGV5opXmhRdIlSgC05TCamWjNNQVmFg2KAHJY-RU6g-15QA)

Reply:
Sure, both makes sense. I just suggest leaving a comment for the intent of the comparison so the future us know that is an intended `!=`.